cli/cmd/compliance.go

```diff
@@ -213,6 +214,90 @@ func complianceReportRecommendationsTable(recommendations []api.ComplianceRecomm
 	return out
 }
 
+func complianceCSVReportAddRecord(
+	tenantAlias string,
+	accountAlias string,
+	reportType string,
+	reportTime time.Time,
+	recommendation api.ComplianceRecommendation,
+	resource string,
+	region string,
+	reasons []string,
+	suppressed bool,
+) []string {
+	record := []string{reportType, reportTime.Format(time.RFC3339)}
+	status := recommendation.Status
+	if suppressed {
+		status = "Suppressed"
+	}
+
+	if tenantAlias != "" {
+		record = append(record, tenantAlias)
+	}
+
+	record = append(record, []string{
+		accountAlias,
+		recommendation.Category,
+		recommendation.RecID,
+		recommendation.Title,
+		status,
+		recommendation.SeverityString(),
+		resource,
+		region,
+		strings.Join(reasons, ","),
+	}...)
+
+	return record
+}
+
+func complianceCSVReportRecommendationsTable(
+	tenantAlias string,
+	accountAlias string,
+	reportType string,
+	reportTime time.Time,
+	recommendations []api.ComplianceRecommendation) [][]string {
+	out := [][]string{}
```

Comment:
I like the idea to make this generic across the 3 clouds. But rather than having param name that don't relate when the function is called (eg. when called from compliace_aws.go there is no tenant alias) Would it be possible to create a struct container these values and pass it in?

Reply:
Of course! I rework this.